commcare_connect/opportunity/tasks.py

```diff
@@ -284,10 +284,11 @@ def bulk_approve_completed_work():
         access.payment_accrued = 0
         for completed_work in completed_works:
             approved_count = completed_work.approved_count
-            visits = completed_work.uservisit_set.values_list("status", flat=True)
-            if any(visit == "rejected" for visit in visits):
+            visits = completed_work.uservisit_set.values_list("status", "reason")
+            if any(status == "rejected" for status, _ in visits):
                 completed_work.status = CompletedWorkStatus.rejected
-            elif all(visit == "approved" for visit in visits):
+                completed_work.reason = "\n".join(map(lambda v: v[1], filter(lambda v: bool(v[1]), visits)))
```

Comment:
I approved but I think this code would be much clearer as a list comprehension
`"\n".join([v[1] for v in visits if v[1]])`

Reply:
Done. https://github.com/dimagi/commcare-connect/pull/318/commits/b170f8e1f60b5fa9eae9536226c56bd50c6ecd9f